application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/AttachmentListener.java

```diff
@@ -0,0 +1,118 @@
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public AttachmentListener(@NotNull Config config,
+            @NotNull ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.config = config;
+        this.modAuditLogWriter = modAuditLogWriter;
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (attachmentContainsBlacklistedFileExtension(event.getMessage())) {
+            deleteMessageAndDmUser(event);
+            warnMods(event);
+        }
+    }
+
+    private void deleteMessageAndDmUser(@NotNull MessageReceivedEvent event) {
+        event.getMessage().delete().flatMap(any -> dmUser(event)).queue();
+    }
+
+    private RestAction<Message> dmUser(@NotNull MessageReceivedEvent event) {
+        Message dmMessage = createDmMessage(event.getMessage());
+        return event.getJDA()
+            .openPrivateChannelById(event.getAuthor().getIdLong())
+            .flatMap(channel -> channel.sendMessage(dmMessage));
+    }
+
+    private Message createDmMessage(Message originalMessage) {
+        String originalMessageContent = originalMessage.getContentDisplay();
+        String blacklistedAttachments = getBlacklistedAttachmentsFromMessage(originalMessage);
+        String dmMessageContent =
+                "Hey there, you posted a message with a blacklisted file attachment: %s. Following file extension are blacklisted: %s ."
+                    .formatted(blacklistedAttachments, config.getBlacklistedFileExtensions());
+        // No embedded needed if there was no message from the user
+        if (StringUtils.isEmpty(originalMessageContent)) {
+            return new MessageBuilder(dmMessageContent).build();
+        }
+        return createMessageWithOriginalMessageAsEmbedded(originalMessageContent, dmMessageContent);
+    }
+
+    private Message createMessageWithOriginalMessageAsEmbedded(String originalMessageContent,
+            String dmMessageContent) {
+        MessageEmbed originalMessageEmbed =
+                new EmbedBuilder().setDescription(originalMessageContent)
+                    .setColor(Color.ORANGE)
+                    .build();
+        return new MessageBuilder(dmMessageContent).setEmbeds(originalMessageEmbed).build();
+    }
+
+    private String getBlacklistedAttachmentsFromMessage(Message originalMessage) {
+        List<String> blacklistedAttachments = new ArrayList<>();
+        originalMessage.getAttachments()
+            .stream()
+            .filter(attachment -> config.getBlacklistedFileExtensions()
+                .contains(attachment.getFileExtension()))
+            .forEach(attachment -> blacklistedAttachments.add(attachment.getFileName()));
```

Comment:
Instead, you should use the stream-api properly and continue like `.map(Attachment::getFileName).toList()`